lib/sidekiq_unique_jobs/server/middleware.rb

```diff
@@ -67,6 +108,33 @@ def logger
       def connection(&block)
         SidekiqUniqueJobs::Connectors.connection(redis_pool, &block)
       end
+
+      def unlock_run(lock_key, item)
+        connection do |con|
+          con.eval(remove_on_match, keys: ["#{lock_key}:run"], argv: [item['jid']])
+        end
+      end
+
+      def reschedule(item)
+        Sidekiq::Client.new(redis_pool).raw_push([item])
+      end
+
+      # rubocop:disable HandleExceptions
+      def try_acquire_run_lock(lock_key, item)
+        status = begin
+          (run_lock_retries + 1).times do
+            status = connection do |con|
+              con.set("#{lock_key}:run", item['jid'], nx: true, ex: run_lock_expire)
+            end
+            break if status
+            sleep(run_lock_retry_interval)
```

Comment:
A better approach to me would be to push this back for later processing but I haven't delved into the details of the code I just feel that would be a safer approach. It is also a feature that people have asked for being able to just delay a locked job until the currently running one is done. `Reschedulable` jobs I suppose it would be called.